docs/latest/user/udp-routing.md

```diff
@@ -0,0 +1,196 @@
+# UDP Routing
+
+The [UDPRoute](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.UDPRoute/)
+resource allows users to configure UDP routing by matching UDP traffic and forwarding it to
+Kubernetes backends. This guide will use a CoredDNS example to walk you through the steps required to configure
+ UDPRoute on Envoy Gateway.
+
```

Comment:
@zhaohuabing can you please add the point that UDPRoute acts in "non-transparent mode"? For example:

> UDPRoute allows Envoy to operate as a non-transparent proxy between a UDP client and server. The lack of transparency means that the upstream server will see the source IP and port of the Gateway instead of the client. For additional information, refer to Envoy's [UDP proxy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/udp_filters/udp_proxy).